### PR TITLE
Remove path-mapping from uri translator

### PR DESCRIPTION
**Pull request by floitsch**
Also start fixing some LSP tests.

### CODE REVIEW

File: tools/lsp/server/uri_path_translator.toit [hunk 1]
```diff
@@ -60,44 +60,56 @@ percent_decode_ str:
...
+          // Remove the leading '/'.
+          decoded = decoded.trim --left "/"
+      return decoded
     // For every other uri assume that it's stored in the source-bundle and
     // mark it as virtual.
     return "$VIRTUAL_FILE_MARKER_$uri"
 
-  // Returns a canonicalized version of the uri.
+  /**
+  Returns a canonicalized version of the uri.
```
Comment (kasperl) [inline tools/lsp/server/uri_path_translator.toit:104]:
```suggestion
  Returns a canonicalized version of the $uri.
```

or

```suggestion
  Returns a canonicalized version of the URI.
```

File: tools/lsp/server/uri_path_translator.toit [hunk 2]
```diff
@@ -60,44 +60,56 @@ percent_decode_ str:
...
+      // three slashes.
+      // The 'path' is always absolute, and on Linux, we don't add the
+      // additional leading '/' for the root.
+      // However, on Windows, we have to add the leading '/' to start the
+      // path part.
+      path = "/$path"
     assert: path.starts_with "/"
+    encoded := percent_encode_ path
+    return "file://" + encoded
     return path.trim --left "/"
```
Comment (kasperl) [inline tools/lsp/server/uri_path_translator.toit:87]: Drop this.

**Review by kasperl — APPROVED** [review]
Looks good to me.

File: tools/lsp/server/uri_path_translator.toit (hunk 1 above)
Comment (floitsch) [reply tools/lsp/server/uri_path_translator.toit:104]: done. ($uri)

File: tools/lsp/server/uri_path_translator.toit (hunk 2 above)
Comment (floitsch) [reply tools/lsp/server/uri_path_translator.toit:87]: oops.
done.

**Pull request merged**